lib/fauna/query.rb

```diff
@@ -457,15 +457,11 @@ def self.not(boolean)
 
   private
 
-    def self.with_auto_vars(n_vars)
-      low_var_number = Thread.current[:fauna_lambda_var_number] || 0
-      next_var_number = low_var_number + n_vars
-
-      Thread.current[:fauna_lambda_var_number] = next_var_number
-
-      yield (low_var_number...next_var_number).map { |i| "auto#{i}" }
-    ensure
-      Thread.current[:fauna_lambda_var_number] = low_var_number
+    def self.block_parameters(block)
+      block.parameters.map do |kind, name|
+        fail ArgumentError, 'All lambda parameters must be non-rest.' if kind == :rest
```

Comment:
'splat' is the right nomenclature: "Splat parameters are not supported in lambda expressions."